homeassistant/components/vodafone_station/device_tracker.py

```diff
@@ -0,0 +1,127 @@
+"""Support for Vodafone Station routers."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+from aiovodafone import VodafoneStationDevice
+
+from homeassistant.components.device_tracker import ScannerEntity, SourceType
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import _LOGGER, DOMAIN
+from .coordinator import VodafoneStationDeviceInfo, VodafoneStationRouter
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up device tracker for Vodafone Station component."""
+
+    _LOGGER.debug("Start device trackers setup")
+    coordinator: VodafoneStationRouter = hass.data[DOMAIN][entry.entry_id]
+
+    tracked: set = set()
+
+    @callback
+    def async_update_router() -> None:
+        """Update the values of the router."""
+        async_add_new_tracked_entities(coordinator, async_add_entities, tracked)
+
+    entry.async_on_unload(
+        async_dispatcher_connect(
+            hass, coordinator.signal_device_new, async_update_router
+        )
+    )
+
+    async_update_router()
+
+
+@callback
+def async_add_new_tracked_entities(
+    coordinator: VodafoneStationRouter,
+    async_add_entities: AddEntitiesCallback,
+    tracked: set[str],
+) -> None:
+    """Add new tracker entities from the router."""
+    new_tracked = []
+
+    _LOGGER.debug("Adding device trackers entities")
+    for mac, device_info in coordinator.data.devices.items():
+        if mac in tracked:
+            continue
+        _LOGGER.debug("New device tracker: %s", device_info.device.name)
+        new_tracked.append(VodafoneStationTracker(coordinator, device_info))
+        tracked.add(mac)
+
+    async_add_entities(new_tracked)
+
+
+class VodafoneStationTracker(CoordinatorEntity[VodafoneStationRouter], ScannerEntity):
```

Comment:
the downside to using `CoordinatorEntity` and not listening for the signals is that every poll is going to try to write state for every device tracker even if the state hasn't changed

Reply:
This is fine for the first implementation but we should refactor to only dispatch when something changes before more platforms are added